src/sqlfluff/dialects/dialect_spark3.py

```diff
@@ -783,3 +783,117 @@ class JoinClauseSegment(BaseSegment):
     get_eventual_alias = ansi_dialect.get_segment(
         "JoinClauseSegment"
     ).get_eventual_alias
+
+
+@spark3_dialect.segment(replace=True)
+class TableExpressionSegment(BaseSegment):
+    """The main table expression e.g. within a FROM clause."""
+
+    type = "table_expression"
+    match_grammar = OneOf(
+        Ref("ValuesClauseSegment"),
+        Ref("BareFunctionSegment"),
+        Ref("FunctionSegment"),
+        Ref("TableReferenceSegment"),
+        # Nested Selects
+        Bracketed(Ref("SelectableGrammar")),
+    )
+
+
+@spark3_dialect.segment(replace=True)
+class AliasExpressionSegment(BaseSegment):
+    """A reference to an object with an `AS` clause.
+
+    The optional AS keyword allows both implicit and explicit aliasing.
+    Note also that it's possible to specify just column aliases without aliasing the table as well:
+    .. code-block:: sql
+
+        SELECT * FROM VALUES (1,2) as t (a, b);
+        SELECT * FROM VALUES (1,2) as t;
+        SELECT * FROM VALUES (1,2) as (a, b);
+
+    """
+
+    type = "alias_expression"
+    match_grammar = Sequence(
+        Ref.keyword("AS", optional=True),
+        OneOf(
+            Sequence(
+                Ref("SingleIdentifierGrammar", optional=True),
+                # Column alias in VALUES clause
+                Bracketed(Ref("SingleIdentifierListSegment"), optional=True),
+            ),
+            Ref("QuotedLiteralSegment"),
+        ),
+    )
+
+
+@spark3_dialect.segment()
+class DelimitedValues(BaseSegment):
```

Comment:
I started with `DelimitedValues`, `ScalarValue`, and `TupleValue` inlined into `ValuesClauseSegment`, but factored them out for my sanity while debugging and don't see a particular reason to go back to the in-lined version.

Reply:
Sounds good. Someday, if we have rules looking at these specific kind of segment, they may benefit from making things more explicit. Just a few weeks ago, I refactored the `CASE` grammar to create segments for the various parts (`WHEN`, `ELSE`). It made a couple of rules **much** simpler.

Reply:
Are all these new segments still needed in the Spark dialect, now that the ANSI dialect provides `ValuesClauseSegment`?

Reply:
ANSI provided `ValuesClauseSegment` before, it just didn't work because it was trying to parse it as a function call. So I was able to remove that segment from my changes, but I don't think that was the only issue.

 After merging recent changes, I tried:

1. comment out my new segments
2. run dialects tests, observed failures
3. `tox -e generate-fixture-yml`, maybe it parses but we have the wrong expectation
4. run dialects tests, still observe failures
5. uncomment some of my new segements; go to 1 until tests pass or all new segments are required 

Does that seem like it would determine whether the upstream changes "just work" for me?

Reply:
I think someone recently fixed the issue where it would parse as a function call. Was this occurring when there was exactly _one_ column?

@tunetheweb: Do you remember the history about this? IIRC, it was fixed in the last week or so.

Reply:
Found it! See [this PR comment:](https://github.com/sqlfluff/sqlfluff/pull/2404#discussion_r790067975)
>This prevents the values clause from accidently being parsed as a function as described in #2406

Reply:
Going to merge in `main` here to see what it looks like.